app/models/event.rb

```diff
@@ -6,6 +6,13 @@ class Event < ActiveRecord::Base
   validates :start_on, presence: true
   validates :end_on, presence: true
   validates :name, presence: true
+  validate :validate_end_on_before_start_on
+
+  def validate_end_on_before_start_on
+    if end_on && start_on
```

Comment:
After discussing this with my colleagues, we agree this is the best way. 😀 

We think it's to use the `#add` method on the errors object, though:

```
errors.add(:base, "Start date must be before end date")
```